control-plane-operator/controllers/hostedcontrolplane/olm/catalogs.go

```diff
@@ -135,17 +135,38 @@ func GetCatalogImages(ctx context.Context, hcp hyperv1.HostedControlPlane, pullS
 		return nil, fmt.Errorf("invalid OpenShift release version format: %s", imageConfig.Config.Labels["io.openshift.release"])
 	}
 
+	registries := []string{
+		"registry.redhat.io/redhat",
+	}
+	if len(registryOverrides) > 0 {
+		for registrySource, registryDest := range registryOverrides {
+			if registries[0] == registrySource {
+				registries = registryDest
+				break
+			}
+		}
+	}
+
 	//check catalogs of last 4 supported version in case new version is not available
 	supportedVersions := 4
+	imageRegistry := ""
 	for i := 0; i < supportedVersions; i++ {
-		_, err = digestLister(ctx, fmt.Sprintf("registry.redhat.io/redhat/certified-operator-index:v%d.%d", version.Major, version.Minor), pullSecret)
-		if err == nil {
-			break
+		for _, registry := range registries {
+			testImage := fmt.Sprintf("%s/certified-operator-index:v%d.%d", registry, version.Major, version.Minor)
+
```

Comment:
should this use imageMetaDataProvider to pick the right mirrors in the case no annotations are pased through? @jparrill